Jenkinsfiles/Jenkinsfile.cbc-auto-build-deploy-migrate-app-ami

```diff
@@ -0,0 +1,308 @@
+    )
+
+    choice(
+      name: "APP_MIGRATIONS",
+      choices: ["", "Pre-deploy", "Post-deploy", "No Migrations"],
+      description: "Are there any migrations required for this deploy?"
+    )
+
+    string(
+      name: "APP_REPO_BRANCH",
+      defaultValue: "master",
+      description: "The branch of the application repo to build in AMI. Defaults to master."
+    )
+
+    string(
+      name: "DEPLOY_REPO_BRANCH",
+      defaultValue: "master",
+      description: "The branch of the deployment repo to build in AMI. Defaults to master."
+    )
+
+    string(
+      name: "PLATINUM_AMI_ID",
+      defaultValue: "",
+      description: "The AMI ID of the platinum image. If unset, the latest will be pulled from AWS and used."
+    )
+
+    string(
+      name: "APP_AMI_ID",
+      defaultValue: "",
+      description: "The AMI ID for the app image. Leave blank to use AMI from this job's build stage. To use a pre-built AMI instead, put AMI ID here (this will skip the build stage of this job)."
+    )
+    
+  }
+
+  stages {
+    // Migration selection check
+    stage("Check migration status") {
+      steps {
+        script {
+          if (params.APP_MIGRATIONS == '') {
+            currentBuild.result = 'ABORTED'
+            error('No app migration selection was made.')
+          }
+        }
+      }
+    }
+
+    stage("Notify Slack") {
+      steps {
+        script {
+          helpers = load "Jenkinsfiles/helpers.groovy"
+          helpers.slackNotify "STARTING - ENV:${params.APP_ENV}"
+        }
+      }
+    }
+
+    // BUILD
+
+    // TODO:
+    // ** Think about the scenario where this automated job gets aborted after the build stage - 
+    // ** consider an option to use the previously built AMI when running this job again
+    // ** The 'when' condition below will skip the build stage if a value is supplied for the APP_AMI_ID
+    stage("Build this release") {
+      when {
+        expression { params.APP_AMI_ID == "" }
+      }
+      steps {
+        // TODO: Change to production job after testing is complete
+        build job: 'Z BUILD - App AMI (AMZN 2)', 
+          parameters: [
+            string(name: 'APP_REPO_BRANCH', value: params.APP_REPO_BRANCH.toLowerCase()), 
+            string(name: 'DEPLOY_REPO_BRANCH', value: params.DEPLOY_REPO_BRANCH), 
+            string(name: 'PLATINUM_AMI_ID', value: params.PLATINUM_AMI_ID)
+          ]
+      }
+    }
+
+    // Ansible config - Pre-deploy migrations
+    stage("Determine Ansible Config: Pre-deploy") {
+      // TODO: Thinking this will need to be run separately for Pre/Post migrations since instances change post deploy
+      when {
+        expression { params.APP_MIGRATIONS == "Pre-deploy" }
+      }
+      steps {
+        script {
+          echo("Setting ansible config for tagged deployment instance")
+          ANSIBLE_INVENTORY = "inventory/ec2.py"
+          ANSIBLE_SUBSET = "tag_Function_app_AppServer"
```

Comment:
It doesn't look like these parameters change for the entire duration of the job wether they are pre or post deploy, so I'd guess you can set them once at the top of the job in the environment vars section and be done with it.